internal/controller/datadogagentinternal/controller_reconcile_v2_common.go

```diff
@@ -100,7 +100,7 @@ func (r *Reconciler) createOrUpdateDeployment(parentLogger logr.Logger, ddai *da
 		if !needUpdate {
 			// no need to update hasn't changed
 			now := metav1.NewTime(time.Now())
-			updateStatusFunc(deployment, newStatus, now, metav1.ConditionTrue, createSucceeded, "Deployment created")
+			updateStatusFunc(currentDeployment, newStatus, now, metav1.ConditionTrue, createSucceeded, "Deployment created")
```

Comment:
```suggestion
			updateStatusFunc(currentDeployment, newStatus, now, metav1.ConditionTrue, "DeploymentUpToDate", "Deployment up-to-date")
```
Ref: https://github.com/DataDog/datadog-operator/blob/ccc681c668d4f504268158fa3fef91840ce8c070/internal/controller/datadogagent/controller_reconcile_v2_common.go#L111